### ISSUE TITLE
math -T with time

### ISSUE THREAD

**Issue by delaviel-anger** [GenericMappingTools/gmt#4771]
> gmt -v 6.2.0_bb0b2d9_2020.12.14 on macOS Big Sur

When trying to create a vector of timestamps (e.g 1 over 4 Jan 1st from 1980 to 2019)
```
gmt math -o0 -T1980-01-01T/2019-12-31T/4y T =
00T00:00:00.000000000
00T00:00:00.000000000
00T00:00:00.000000000
00T00:00:00.000000000
00T00:00:00.000000000
00T00:00:00.000000000
00T00:00:00.000000000
00T00:00:00.000000000
00T00:00:00.000000000
00T00:00:00.000000000
```

However the *inc***d** version works :
```
gmt math -o0 -T1980-01-01T/2019-12-31T/1461d T =
1980-01-01T00:00:00
1984-01-01T00:00:00
1988-01-01T00:00:00
1992-01-01T00:00:00
1996-01-01T00:00:00
2000-01-01T00:00:00
2004-01-01T00:00:00
2008-01-01T00:00:00
2012-01-01T00:00:00
2016-01-01T00:00:00
```
After some tests, it seems that only `d` is properly working.
`w` acts like seconds
`o` and `y` have the output bug

Thanks :)

**Comment by meghanrjones** [comment]
This bug was introduced by this [merge commit](https://github.com/GenericMappingTools/gmt/commit/3ebebec8554f7368f8300ff5f169901b096be357) (PR: https://github.com/GenericMappingTools/gmt/pull/4409). I'll take a look and will ping for help if needed.

**Comment by meghanrjones** [comment]
Fixed by #4778

**Comment by meghanrjones** [comment]
>  `w` acts like seconds

Turns out weeks isn't supported, #4778 updates the documentation accordingly.
